### PR TITLE
docs(Datepicker): Add custom format examples

### PR DESCRIPTION
**Pull request by pompomon**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Adding example usage for custom formatting and parsing props for `Datepicker` and visual tests

### CODE REVIEW

File: packages/fluentui/docs/src/examples/components/Datepicker/Usage/DatepickerFormatExample.shorthand.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Datepicker } from '@fluentui/react-northstar';
+import * as React from 'react';
+
+const DatepickerFormatExample = () => {
+  const customFormat = (date: Date) => {
+    const day = date.getDate();
+    const year = date.getFullYear();
+    return `${day} of ${year}`;
+  };
+  return <Datepicker formatMonthDayYear={customFormat} today={new Date(2020, 8, 12, 0, 0, 0, 0)} />;
```
Comment (assuncaocharles) [inline packages/fluentui/docs/src/examples/components/Datepicker/Usage/DatepickerFormatExample.shorthand.tsx:10]: Just curiosity `formatMonthDayYear` works with string patterns also? e.g `YY/MM/DD`

Comment (pompomon) [reply packages/fluentui/docs/src/examples/components/Datepicker/Usage/DatepickerFormatExample.shorthand.tsx:10]: Well, that would be up to the consumer, like they can have `customFormat` method defined like that:

```  const customFormat = (date: Date) => {
    const options = {
      weekday: 'short',
      year: 'numeric',
      month: 'short',
      day: '2-digit',
    };

    return date.toLocaleDateString('en-US', options);
  };
```

File: packages/fluentui/docs/src/examples/components/Datepicker/Usage/DatepickerFormatExample.shorthand.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Datepicker } from '@fluentui/react-northstar';
+import * as React from 'react';
+
+const DatepickerFormatExample = () => {
+  const customFormat = (date: Date) => {
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Datepicker/Usage/DatepickerFormatExample.shorthand.tsx:5]:
```suggestion
  const customFormatter: SOME_TYPE = (date) => {
```

Does thing has typings? Can we use them?

File: packages/fluentui/docs/src/examples/components/Datepicker/Usage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+
+import ComponentExample from '../../../../components/ComponentDoc/ComponentExample';
+import ExampleSection from '../../../../components/ComponentDoc/ExampleSection';
+
+const Usage = () => (
+  <ExampleSection title="Usage">
+    <ComponentExample
+      title="Custom formatting"
+      description="Custom date formatting"
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Datepicker/Usage/index.tsx:10]:
```suggestion
      description={<><code>formatMonthDayYear</code> prop can be used for custom date formatting in PLACE WHERE THIS DATE WILL BE VISIBLE</>}
```

??

File: packages/fluentui/docs/src/examples/components/Datepicker/Usage/index.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      examplePath="components/Datepicker/Usage/DatepickerFormatExample"
+    />
+    <ComponentExample
+      title="Custom parsing"
+      description="Custom date parsing"
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Datepicker/Usage/index.tsx:15]:
```suggestion
      description={<><code>parseDate</code> prop can be used to parse dates that came from user's input</>}
```

?

File: packages/fluentui/docs/src/examples/components/Datepicker/Usage/DatepickerParseExample.shorthand.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Datepicker } from '@fluentui/react-northstar';
+import * as React from 'react';
+
+const DatepickerParseExample = () => {
+  const customParse = (dateString: string) => {
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Datepicker/Usage/DatepickerParseExample.shorthand.tsx:5]: The same question about typings 😺

**Pull request merged**